hw/ip/entropy_src/doc/_index.md

```diff
@@ -10,83 +10,143 @@ This module conforms to the [Comportable guideline for peripheral functionality.
+By default, all tests are enabled, but can be turn off in the {{< regref "CONF" >}} register.
+Because of the variability of the noise source, there are several registers that log statistics associated with the health tests.
+For example, the adaptive proportion test has a high watermark register that logs the highest measured number of ones.
+The {{< regref "ADAPTP_HI_WATERMARKS" >}} register has an entry for both normal and boot-time modes.
+This register allows for determining how close the threshold value should be set to the fail over value.
+Specific to the adaptive proportion test, there is also the {{< regref "ADAPTP_LO_WATERMARKS" >}} register, which will hold the lowest number of ones measured.
+To help understand how well the thresholds work through time, a running count of test fails is kept in the {{< regref "ADAPTP_HI_TOTAL_FAILS" >}} register.
+The above example for the adaptive proportion test also applies to the other health tests, with the exception of the low watermark registers.
+See the timing diagrams below for more details on how the health tests work.
+
+The Mailbox test and Vendor-specific tests are planed, but not yet implemented.
+The theory of operation for these tests will therefore be described in a future update.
+
+The {{< regref "ALERT_THRESHOLD" >}} register determines how many fails can occur before an alert is issued.
+By default, the current threshold is set to two, such that the occurrence of two failing test cycles back-to-back would provide a very low &alpha; value.
+The {{< regref "ALERT_FAIL_COUNTS" >}} register holds the total number of fails, plus all of the individual contributing failing tests.
+Setting the {{< regref "ALERT_THRESHOLD" >}} register to zero will disable alert generation.
+
+Firmware has a path to read entropy from the ENTROPY_SRC block.
+The {{< regref "ENTROPY_CONTROL" >}} register allows firmware to set the internal multiplexers to steer entropy data to the {{< regref "ENTROPY_DATA" >}} register.
+The control bit `ES_TYPE` sets whether the entropy will come from the conditioning block or be sourced through the bypass path.
+A status bit will be set that can either be polled or generate an interrupt when the entropy bits are available to be read from the {{< regref "ENTROPY_DATA" >}} register.
+The firmware needs to read the {{< regref "ENTROPY_DATA" >}} register twelve times in order to cleanly evacuate the 384-bit seed from the hardware path.
+The firmware will directly read out of the main entropy FIFO, and when the control bit `ES_ROUTE` is set, no entropy is being passed to the block hardware interface.
+If the main entropy FIFO fills up, additional entropy that has been health checked and conditioned will be dropped at that point.
+
+The above process will be repeated for as long as entropy bits are to be collected and processed.
+
+At any time, the `ENABLE` field can be cleared to halt the entropy generation (and health check testing) immediately.
```

Comment:
@zi-v: This is how you create on-demand entropy.  Also we will add a interrupt to CSRNG as part of #3575 so that CSRNG can ping firmware to turn the entropy_src back on or off.